### DISCUSSION TITLE
Can Martketplace templates be on a private Git provider?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#410] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12.2.1

### Description

Hi!

When configuring a new Marketplace template, I need to provide the repository URL where the code of my template is. Can I have this repository in a private Git provider? If yes, how can I configure the Console to authenticate with my Git Provider? I haven't found much in documentation about this topic.

Thanks

**Comment by frengooo** [comment, marked as answer]
Hi @Lucascanna,
it should be possible through the Marketplace Provider, that you can find here:
[https://docs.mia-platform.eu/docs/console/company-configuration/providers/marketplace](https://docs.mia-platform.eu/docs/console/company-configuration/providers/marketplace).

Using the Marketplace Provider you can insert the data of a private Git Provider and download templates from it.
Remember to modify the "Marketplace" item too, on the related mongodb collection, inserting the name of the marketplace provider to be used.

Tell me if this works.

**Comment by pmarchini** [comment]
Ciao @Lucascanna, 

Short answer: yes you can 🚀 

You need to define, inside **marketplace item**, the `providerId` of the **GitProvider** that has access to this repository!
You can find more detail regarding the `providerId` property [here](https://docs.mia-platform.eu/docs/marketplace/add_to_marketplace/contributing_overview#common-to-microservice-items-plugins-templates-examples).

If you need to allow other companies to use this template then you will have to create a [**Marketplace Provider**](https://docs.mia-platform.eu/docs/console/company-configuration/providers/marketplace) with access to the private repository in object.

Hope this solve your issue!!!

Let me know!
